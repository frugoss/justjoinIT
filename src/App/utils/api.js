export let API_HOST = 'http://localhost:7000'
if (process.env.NODE_ENV === 'test') {
  API_HOST = 'http://localhost'
}
