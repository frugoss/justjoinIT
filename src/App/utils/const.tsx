

export interface userInterface {
    auth?: boolean,
    name?: string,
    userID?: string,
    loggedPopup?: boolean,
    offPopup?: boolean,
    createPopup?: boolean,
    addPopup?: boolean
}


export interface offerInterface {
    _id: string,
    agreements: string,
    city: string,
    company: string,
    companyIndustry: string,
    companySize: string,
    coordinates: [{ lat: number, lng: number }],
    currency: string,
    description: string,
    email: string,
    employment: string,
    experience: string,
    logo: string | undefined,
    maxSalary: number,
    minSalary: number,
    street: string,
    techStack: [{ language: string, lvl: number }],
    technology: [{ color: string, background: string, img: string, name: string }],
    title: string,
    userID: string,
    website: string,
}

export interface formInterface {
    _id?: string,
    agreements?: string,
    city?: string,
    company?: string,
    companyIndustry?: string,
    companySize?: string,
    coordinates?: [{ lat: number, lng: number }],
    currency?: string,
    description?: string,
    email?: string,
    employment?: string,
    experience?: string,
    logo?: string| undefined,
    maxSalary?: number,
    minSalary?: number,
    street?: string,
    techStack: { language?: string, lvl?: number }[],
    technology?: [{ color: string, background: string, img: string, name: string }],
    title?: string,
    userID: string,
    website?: string,
}


export interface applicationInterface {
    checkbox: boolean,
    email: string,
    file: string,
    message: string,
    name: string,
    offerID: string
}

export interface filtersInterface{
    city: string,
    language: string,
    experience: string,
    salarymax: number,
    salarymin: number,
}

export interface matchInterface{
    city: string,
    language: string,
    experience: string,
    salarymax: string,
    salarymin: string,
}

export interface formValuesCV  {
    name:string,
    email:string,
    message: string,
    checkbox: string,
    file: File,
    offerID: string
}

export interface languageInterface {
    name: string,
    to: string,
    img: string,
    color: string,
    background: string
}



export interface mobileViewModeInterface {
    view: string,
    header: boolean
}

export interface cityInterface {
    name: string,
    to: string,
    coordinates: {lat:number, lng:number}
}
