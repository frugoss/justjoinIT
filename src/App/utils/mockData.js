export const offersList = [
  {
    "_id": "5e9cfa222f9c0e0965b86034",
    "userID": "5e8cc6d853004b069507091d",
    "technology": [
      {
        "img": "data:image/png;baC",
        "color": "rgb(255, 199, 6)",
        "name": "javascript",
        "background": "rgb(255, 199, 6), rgb(255, 175, 0)"
      }
    ],
    "techStack": [
      {
        "language": "JavaScript",
        "lvl": 5
      },
      {
        "language": "ES7",
        "lvl": 3
      }
    ],
    "logo": "data:image/png;basI=",
    "coordinates": [
      {
        "lat": 54.4399555,
        "lng": 18.5440814
      }
    ],
    "city": "Sopot",
    "street": "Tuwima 20",
    "agreements": "Informujemy, że administratorem danych jest dddd z siedzibą w ___________, ul. _______________ (dalej jako \"administrator\"). Masz prawo do żądania dostępu do swoich danych osobowych, ich sprostowania, usunięcia lub ograniczenia przetwarzania, prawo do wniesienia sprzeciwu wobec przetwarzania, a także prawo do przenoszenia danych oraz wniesienia skargi do organu nadzorczego. Dane osobowe przetwarzane będą w celu realizacji procesu rekrutacji. Podanie danych w zakresie wynikającym z ustawy z dnia 26 czerwca 1974 r. Kodeks pracy jest obowiązkowe. W pozostałym zakresie podanie danych jest dobrowolne. Odmowa podania danych obowiązkowych może skutkować brakiem możliwości przeprowadzenia procesu rekrutacji. Administrator przetwarza dane obowiązkowe na podstawie ciążącego na nim obowiązku prawnego, zaś w zakresie danych dodatkowych podstawą przetwarzania jest zgoda. Dane osobowe będą przetwarzane do czasu zakończenia postępowania rekrutacyjnego i przez okres możliwości dochodzenia ewentualnych roszczeń, a w przypadku wyrażenia zgody na udział w przyszłych postępowaniach rekrutacyjnych - do czasu wycofania tej zgody. Zgoda na przetwarzanie danych osobowych może zostać wycofana w dowolnym momencie. Odbiorcą danych jest serwis Just Join IT oraz inne podmioty, którym powierzyliśmy przetwarzanie danych w związku z rekrutacją.",
    "company": "Travers",
    "companyType": "Software House",
    "companyIndustry": "Medicine",
    "currency": "PLN",
    "employment": "Mandate Contract",
    "experience": "Senior",
    "title": "Javascript developer",
    "minSalary": 15000,
    "maxSalary": 25000,
    "description": "<div><!--block--><strong>Who we are looking for<br></strong><br></div><ul><li><!--block--><strong>Senior Angular Developer (3+ experience)</strong></li><li><!--block-->Good analytical skills (problem solving)</li><li><!--block-->Good communication skills</li><li><!--block-->Enjoy helping other Engineers improve and become better</li><li><!--block-->Very good English speaking skills</li></ul><div><!--block--><strong><br>What we offer<br></strong><br></div><ul><li><!--block-->Working in a Development Team (4 team members) alongside backend team placed in London</li><li><!--block-->Lead a team of talented and collaborative engineers&nbsp;</li><li><!--block-->Help mentor and coach members of the team</li><li><!--block-->Help evolve development standards and practices</li><li><!--block-->Review Pull Requests from other developers</li><li><!--block-->Overall responsible for the technical quality of what the team is producing</li><li><!--block-->Collaborate with back-end developers, UX and UI designers to improve our products and features</li></ul><div><!--block--><strong><br>Benefits<br></strong><br></div><ul><li><!--block-->26 days of paid holidays per year (+ additionally paid bank holidays)</li><li><!--block-->30 days sick leave per year (paid 80%)</li><li><!--block--><strong>Flexible working hours</strong></li><li><!--block-->Work from home (1 day per week)</li><li><!--block--><strong>Taking part in development-related events</strong> (conferences, trainings)</li><li><!--block-->Private medical care</li><li><!--block-->MultiSport Plus Package</li></ul>",
    "companySize": "10 - 40",
    "website": "www.traversy.com"
  },
  {
    "_id": "5e9a411cfe04a70747f1a7d1",
    "userID": "5e8cc6d853004b069507091d",
    "technology": [
      {
        "img": "data:image/png;base64",
        "color": "rgb(31, 94, 182)",
        "name": "python",
        "background": "rgb(31, 94, 182), rgb(31, 123, 196)"
      }
    ],
    "techStack": [
      {
        "language": "Javascript",
        "lvl": 3
      },
      {
        "language": "C#",
        "lvl": 2
      }
    ],
    "logo": "data:image/png;",
    "companyType": "Software House",
    "companySize": "10",
    "companyIndustry": "E-commerce",
    "experience": "Mid",
    "title": "Python Developer",
    "employment": "Permanent",
    "minSalary": 3000,
    "maxSalary": 10000,
    "currency": "EUR",
    "description": "<div><!--block-->Lorem ipsum dolor sit amet, consectetur adipiscing elit. In tincidunt maximus sem vitae cursus. Integer ullamcorper quam quis molestie dictum. Nam eros augue, aliquam vitae tristique in, laoreet ac massa. Suspendisse ultricies vulputate elit, scelerisque pellentesque turpis tempor in. Donec neque est, feugiat sit amet tempor ut, gravida sit amet ligula. Donec elit ante, pharetra ac ex a, malesuada molestie velit. Vivamus cursus risus mi, quis maximus justo convallis vitae.  Nullam ultrices porttitor quam, non cursus risus convallis vel. Mauris feugiat lectus eu arcu tempor, eget hendrerit eros imperdiet. Ut vitae sodales nisi. Nam eu dolor eu libero tincidunt maximus non vitae odio. Cras neque orci, dignissim eu feugiat ac, consectetur ut nisl. Maecenas euismod sem orci. Phasellus eget nulla et sapien vulputate mollis vel tempor sem. Nulla cursus pharetra lacus, lobortis porta nulla eleifend et. Curabitur quis arcu id velit gravida euismod vitae eu turpis. In suscipit, lectus vitae maximus iaculis, mauris massa porttitor est, vel porttitor orci massa ut quam. Morbi in sapien sit amet nisl faucibus posuere.  Quisque sit amet finibus leo. Aenean placerat sit amet nulla rutrum auctor. Curabitur scelerisque eleifend purus, quis bibendum lectus venenatis vitae. Quisque pellentesque lorem ex, eu facilisis magna convallis eget. Aliquam eget fermentum augue. Ut sollicitudin dui elit, eu consectetur mi rutrum in. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Nam at massa interdum, feugiat augue ac, fringilla mauris. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus ut velit vulputate nulla luctus aliquet.  Sed ornare viverra felis vitae tincidunt. Pellentesque hendrerit massa sit amet nibh viverra lacinia. Duis vitae metus vel libero tristique euismod. Sed congue consequat purus, quis convallis ex venenatis id. Donec lacus massa, tincidunt vel erat quis, accumsan euismod erat. Curabitur leo dolor, aliquet eget fringilla a, fermentum pulvinar nisl. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos.  Proin eleifend sodales ligula in hendrerit. Nam volutpat, enim id varius bibendum, metus orci venenatis nibh, nec vestibulum ex dolor sit amet ipsum. Cras blandit enim vitae justo lobortis, vel efficitur sem luctus. Duis non dolor id erat consectetur eleifend. Nullam ultrices dictum lacus faucibus aliquet. Vestibulum eget diam cursus, fermentum turpis id, aliquet odio. Quisque vitae bibendum nibh, faucibus malesuada mi. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Mauris dictum risus vel metus convallis, vitae cursus metus faucibus. Cras placerat rhoncus enim non tempor. Integer augue odio, efficitur non diam et, luctus lobortis mi. Suspendisse laoreet vestibulum ex non pretium. Nullam rhoncus, risus nec placerat laoreet, dui neque ultrices metus, vitae congue ligula quam a magna. Nullam sodales lacus in aliquet sagittis.</div>",
    "city": "Gdańsk",
    "street": "Tuwima",
    "coordinates": [
      {
        "lat": 54.3659778,
        "lng": 18.6342856
      }
    ],
    "email": "1234@gmail.com",
    "agreements": "Informujemy, że administratorem danych jest _____xDD_________ z siedzibą w ___________, ul. _______________ (dalej jako \"administrator\"). Masz prawo do żądania dostępu do swoich danych osobowych, ich sprostowania, usunięcia lub ograniczenia przetwarzania, prawo do wniesienia sprzeciwu wobec przetwarzania, a także prawo do przenoszenia danych oraz wniesienia skargi do organu nadzorczego. Dane osobowe przetwarzane będą w celu realizacji procesu rekrutacji. Podanie danych w zakresie wynikającym z ustawy z dnia 26 czerwca 1974 r. Kodeks pracy jest obowiązkowe. W pozostałym zakresie podanie danych jest dobrowolne. Odmowa podania danych obowiązkowych może skutkować brakiem możliwości przeprowadzenia procesu rekrutacji. Administrator przetwarza dane obowiązkowe na podstawie ciążącego na nim obowiązku prawnego, zaś w zakresie danych dodatkowych podstawą przetwarzania jest zgoda. Dane osobowe będą przetwarzane do czasu zakończenia postępowania rekrutacyjnego i przez okres możliwości dochodzenia ewentualnych roszczeń, a w przypadku wyrażenia zgody na udział w przyszłych postępowaniach rekrutacyjnych - do czasu wycofania tej zgody. Zgoda na przetwarzanie danych osobowych może zostać wycofana w dowolnym momencie. Odbiorcą danych jest serwis Just Join IT oraz inne podmioty, którym powierzyliśmy przetwarzanie danych w związku z rekrutacją.",
    "website": "www.lion.com",
    "company": "Lion",
    "remote": true
  },
  {
    "_id": "5e9cf7062f9c0e0965b86029",
    "userID": "5e8cc6d853004b069507091d",
    "technology": [
      {
        "img": "data:image/png;base64",
        "color": "rgb(47, 207, 187)",
        "name": "c",
        "background": "rgb(47, 207, 187), rgb(55, 221, 156)"
      }
    ],
    "techStack": [
      {
        "language": "C",
        "lvl": 4
      },
      {
        "language": "Java",
        "lvl": 2
      },
      {
        "language": "Python",
        "lvl": 5
      }
    ],
    "logo": "data:image/jpeg;base64,",
    "currency": "USD",
    "companyType": "Startup",
    "companyIndustry": "Military",
    "email": "test",
    "employment": "B2B",
    "experience": "Mid",
    "city": "Gdynia",
    "coordinates": [
      {
        "lat": 54.51889,
        "lng": 18.53188
      }
    ],
    "street": "Hubala 10",
    "description": "<div><!--block-->The projects, you will work with, come from the most creative and innovative companies in the Silicon Valley, CA, USA and Western Europe. Take part in the things you have read about on the Internet and do the outstanding stuff here: in Poland. The real adventure is waiting for you.<br><br>About project:<br>Recognition of objects from drone photos, processing into 3D models and providing the user with platform to view and analize the results.<br><br></div><div><!--block-->Expected experience:</div><ul><li><!--block-->Very good spoken and written English</li><li><!--block-->Very good knowledge of Python or Node.JS</li><li><!--block-->Good understanding of Google Cloud, Circle CI and Kuberenetes for CI/CD</li><li><!--block-->Good knowledge of SQL and non-SQL databases</li><li><!--block-->Easy to work with</li><li><!--block-->Ambitious and excited to do new stuff</li></ul><div><!--block--><br>Appreciated knowledge:</div><ul><li><!--block-->Good practices in embedded software development</li><li><!--block-->Good understanding of microservices</li><li><!--block-->Experience in leading a small developers team</li></ul><div><!--block-->&nbsp;</div><div><!--block-->We offer:</div><ul><li><!--block-->International cutting edge projects – work with the technology leaders from CA, USA!</li><li><!--block-->Very flexible working hours, home-office during selected days possible</li><li><!--block-->Corporate-free environment, straightforward, simple, easy</li><li><!--block-->Long-term commitment from apptimia side</li></ul>",
    "title": "C Developer",
    "company": "Fine",
    "minSalary": 3000,
    "maxSalary": 5500,
    "companySize": "20 - 10 ",
    "website": "www.fine.com",
    "agreements": "Informujemy, że administratorem danych jest xD z siedzibą w ___________, ul. _______________ (dalej jako \"administrator\"). Masz prawo do żądania dostępu do swoich danych osobowych, ich sprostowania, usunięcia lub ograniczenia przetwarzania, prawo do wniesienia sprzeciwu wobec przetwarzania, a także prawo do przenoszenia danych oraz wniesienia skargi do organu nadzorczego. Dane osobowe przetwarzane będą w celu realizacji procesu rekrutacji. Podanie danych w zakresie wynikającym z ustawy z dnia 26 czerwca 1974 r. Kodeks pracy jest obowiązkowe. W pozostałym zakresie podanie danych jest dobrowolne. Odmowa podania danych obowiązkowych może skutkować brakiem możliwości przeprowadzenia procesu rekrutacji. Administrator przetwarza dane obowiązkowe na podstawie ciążącego na nim obowiązku prawnego, zaś w zakresie danych dodatkowych podstawą przetwarzania jest zgoda. Dane osobowe będą przetwarzane do czasu zakończenia postępowania rekrutacyjnego i przez okres możliwości dochodzenia ewentualnych roszczeń, a w przypadku wyrażenia zgody na udział w przyszłych postępowaniach rekrutacyjnych - do czasu wycofania tej zgody. Zgoda na przetwarzanie danych osobowych może zostać wycofana w dowolnym momencie. Odbiorcą danych jest serwis Just Join IT oraz inne podmioty, którym powierzyliśmy przetwarzanie danych w związku z rekrutacją."
  }
]

export const CV = [
  {
    "name": "Sylwester Kowalski",
    "email": "sylwesterkowalski@gmail.com",
    "message": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sit amet convallis massa. Curabitur efficitur pretium quam ut laoreet. Ut mollis ut velit quis semper. Nam placerat ullamcorper sapien vel laoreet. Vivamus lobortis elit id massa porta condimentum. Curabitur quis nulla eu purus egestas vestibulum. Vivamus semper arcu et blandit faucibus. Fusce sit amet massa cursus, tincidunt leo ut, molestie ligula. Proin sed nisi aliquet, facilisis velit at, scelerisque nunc. Integer sapien leo, porta non pellentesque id, aliquet in magna. Vestibulum venenatis porttitor enim a consequat. Maecenas sed malesuada dolor. Vivamus facilisis laoreet sapien, eu tincidunt dolor bibendum vitae. Etiam id sodales est. Phasellus consequat quam eu nunc tempor auctor.",
    "checkbox": false,
    "offerID": "5e9cfa222f9c0e0965b86034",
    "file": "9fce5a67910bd4d71cb75a400.pdf"
  },
  {
    "name": "Kinga Sylwestrzak",
    "email": "kingasylwestrzak@gmail.com",
    "message": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sit amet convallis massa. Curabitur efficitur pretium quam ut laoreet. Ut mollis ut velit quis semper. Nam placerat ullamcorper sapien vel laoreet. Vivamus lobortis elit id massa porta condimentum. Curabitur quis nulla eu purus egestas vestibulum. Vivamus semper arcu et blandit faucibus. Fusce sit amet massa cursus, tincidunt leo ut, molestie ligula. Proin sed nisi aliquet, facilisis velit at, scelerisque nunc. Integer sapien leo, porta non pellentesque id, aliquet in magna. Vestibulum venenatis porttitor enim a consequat. Maecenas sed malesuada dolor. Vivamus facilisis laoreet sapien, eu tincidunt dolor bibendum vitae. Etiam id sodales est. Phasellus consequat quam eu nunc tempor auctor.",
    "checkbox": true,
    "offerID": "5e9cfa222f9c0e0965b86034",
    "file": "9fce5a67910bd4d71cb75a401.pdf"
  }
]
