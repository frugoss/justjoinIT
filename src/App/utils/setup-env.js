import {server} from 'App/utils/server.js'
beforeAll(() => server.listen())
afterEach(() => server.resetHandlers())
afterAll(() => server.close())
