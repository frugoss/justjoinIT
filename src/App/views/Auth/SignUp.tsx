import React, {useState} from 'react';
import styles from './auth.module.scss';
import logo from '../../assets/images/signUp.svg';
import Button from '@material-ui/core/Button';
import facebook from '../../assets/images/facebookLogo.png';
import github from '../../assets/images/gitLogo.png';
import google from '../../assets/images/googleLogo.png';
import linked from '../../assets/images/linkedinLogo.png';
import SocialButton from './/SocialButton';
import Inputs from './Inputs'
import {Link, RouteComponentProps, withRouter} from "react-router-dom";
import Alert from "@material-ui/lab/Alert";
import Hidden from "@material-ui/core/Hidden";
import {userInterface} from "../../utils/const";


interface SignUpProps extends RouteComponentProps {
    user: userInterface,
    setUser: React.Dispatch<React.SetStateAction<userInterface>>,
}

const SignUp: React.FC<SignUpProps> = ({history, user, setUser}) => {
    const [values, setValues] = useState({
        email: "",
        password: ''
    });
    const [formError, setFormError] = useState("");

    const register = async (event: React.FormEvent<HTMLFormElement>) => {
        event.preventDefault();
        try {
            const response = await fetch("http://192.168.10.25:7000/devs/signup", {
                method: "POST",
                body: JSON.stringify(values),
                headers: {"Content-Type": "application/json"},
                credentials: "include"
            })
            if (response.ok) {
                setUser({...user, createPopup: true})
                history.push("/devs");
            } else {
                response.text()
                    .then(text => setFormError(text));
            }
        } catch (err) {
        }
    };
    return (
        <>
        <div className={styles.row}>
            <div className={styles.column}>
                <h1>Get started for free</h1>
                <div className={styles.center}>
                <div className={styles.socials}>
                    <SocialButton icon={google}>Register with Google </SocialButton>
                    <SocialButton icon={github}>Register with Github </SocialButton>
                </div>
                <div className={styles.socials}>
                    <SocialButton icon={linked}>Register with LinkedIn </SocialButton>
                    <SocialButton icon={facebook}>Register with Facebook </SocialButton>
                </div>
                </div>
                <div className={styles.row}>
                    <hr className={styles.double}/>
                    <span style={{fontWeight: 300}}>Or</span>
                    <hr className={styles.double}/>
                </div>
                <form onSubmit={register}>
                    <Inputs error={formError} values={values} setValues={setValues}/>
                    {formError ? <Alert style={{width:396, marginBottom: 15}} variant="filled" severity="error">
                            {formError} </Alert> : ""}
                    <div className={styles.bottomSign}>
                        <Button type="submit" className={styles.btn} variant="contained" color="secondary">Register</Button>
                        <span>Already have an account?<Link to="/devs"
                                                            className={styles.register}> Sign In</Link></span>
                        <hr className={styles.one}/>
                        <span><a className={styles.register} href="www.google.com">Forgot Password?</a></span>
                    </div>
                </form>
            </div>
            <Hidden smDown>
                <div className={styles.imgContainer}>
                <img alt="sign up" className={styles.img} src={logo}/>
                </div>
            </Hidden>
        </div>
            </>
    );
}


export default withRouter(SignUp);
