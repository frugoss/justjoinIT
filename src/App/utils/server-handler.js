import {rest} from 'msw'
import {offersList, CV} from 'App/utils/mockData.js'

const handlers = [
  rest.post('/devs', async (req, res, ctx) => {
    return res(ctx.text("4325325325235"))
  }),
  rest.post('/devs/signup', async (req, res, ctx) => {
    return res(ctx.text("4325325325235"))
  }),
  rest.get('/', async (req, res, ctx) => {
    return res(ctx.json(offersList))
  }),
  rest.get('/cv/get', async (req, res, ctx) => {
    return res(ctx.json(CV))
  })
]

export {handlers}
