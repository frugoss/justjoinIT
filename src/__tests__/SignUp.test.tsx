import React from 'react';
import "@testing-library/jest-dom/extend-expect"
import SignUp, {SignUpProps} from "../App/views/Auth/SignUp";
import {screen, waitFor} from '@testing-library/react';
import userEvent from "@testing-library/user-event";
import {renderWithRouter} from "../App/utils/tests";
import {server, rest} from "../App/utils/server.js"


function renderSignUp(props: Partial<SignUpProps> = {}) {
    const defaultProps: SignUpProps = {
        user: {},
        setUser: () => {},
        history: {} as any,
        location: {} as any,
        match: {} as any,
    }
    return renderWithRouter(<SignUp {...defaultProps} {...props} />, {route: '/devs/signup'});
}
test("Test if it gets redirected after successful creation of an account",  async () => {
    const {history} = renderSignUp();
    userEvent.type(screen.getByRole("textbox", {name: /email/i}), 'unique@gmail.com');
    userEvent.type(screen.getByLabelText("Password"), '1234');
    userEvent.click(screen.getByRole("button", {name: "Register"}))
    await waitFor( () => expect(history.location.pathname).toEqual('/devs'))


})

test('Snapshot', async () => {
    const {container} = renderSignUp();
    expect(container).toMatchSnapshot();
})
