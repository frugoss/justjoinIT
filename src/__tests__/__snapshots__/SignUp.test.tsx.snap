// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
<div>
  <div
    class="row"
  >
    <div
      class="column"
    >
      <h1>
        Get started for free
      </h1>
      <div
        class="center"
      >
        <div
          class="socials"
        >
          <div
            class="padding"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-11 MuiButton-containedSizeSmall MuiButton-sizeSmall"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                <img
                  alt="logo"
                  class="socialImg"
                  src="googleLogo.png"
                />
                Register with Google 
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="padding"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-11 MuiButton-containedSizeSmall MuiButton-sizeSmall"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                <img
                  alt="logo"
                  class="socialImg"
                  src="gitLogo.png"
                />
                Register with Github 
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="socials"
        >
          <div
            class="padding"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-11 MuiButton-containedSizeSmall MuiButton-sizeSmall"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                <img
                  alt="logo"
                  class="socialImg"
                  src="linkedinLogo.png"
                />
                Register with LinkedIn 
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="padding"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-11 MuiButton-containedSizeSmall MuiButton-sizeSmall"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                <img
                  alt="logo"
                  class="socialImg"
                  src="facebookLogo.png"
                />
                Register with Facebook 
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <hr
          class="double"
        />
        <span
          class="spanWeight"
        >
          Or
        </span>
        <hr
          class="double"
        />
      </div>
      <form>
        <div
          class="inputWidth"
        >
          <div
            class="rowIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 6c1.1 0 2 .9 2 2s-.9 2-2 2-2-.9-2-2 .9-2 2-2m0 10c2.7 0 5.8 1.29 6 2H6c.23-.72 3.31-2 6-2m0-12C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 10c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
            <div
              class="MuiFormControl-root MuiTextField-root makeStyles-margin-13 makeStyles-textField-14"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink="false"
                for="outlined-required"
                id="outlined-required-label"
              >
                Email
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="outlined-required"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-15 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-17"
                  >
                    <span>
                      Email
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="rowIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
              />
            </svg>
            <div
              class="MuiFormControl-root MuiTextField-root makeStyles-margin-13 makeStyles-textField-14"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink="false"
                for="outlined-adornment-password"
                id="outlined-adornment-password-label"
              >
                Password
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                  id="outlined-adornment-password"
                  type="password"
                  value=""
                />
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                >
                  <button
                    aria-label="toggle visibility"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-15 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-17"
                  >
                    <span>
                      Password
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        
        <div
          class="bottomSign"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained btn MuiButton-containedSecondary"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Register
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <span>
            Already have an account?
            <a
              class="register"
              href="/devs"
            >
               Sign In
            </a>
          </span>
          <hr
            class="one"
          />
          <span>
            <a
              class="register"
              href="www.google.com"
            >
              Forgot Password?
            </a>
          </span>
        </div>
      </form>
    </div>
  </div>
</div>
`;
